### PR TITLE
feat(cloud_sql): Add cloud sql psc connectivity samples

### PR DESCRIPTION
**Pull request by isaurabhuttam**
## Description
Added documentation for managing psc connectivity on Cloud SQL instances.

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [X] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [X] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [X] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [X] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [X] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [X] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [X] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s):

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [ ] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [ ] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

**Review by enocom — APPROVED** [review]

File: cloud_sql/instance_psc/main.tf
Comment (rogerthatdev) [inline cloud_sql/instance_psc/main.tf:1]: Please split this sample into 2 separate `main.tf` files (eg `cloud_sql/mysql_instance_psc/main.tf` and `cloud_sql/postgres_instance_psc/main.tf`)

see #443

File: cloud_sql/instance_psc/main.tf [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        allowed_consumer_projects = []
+      }
+      ipv4_enabled = false
+    }
+  }
+}
+# [END cloud_sql_mysql_instance_psc]
+
+# [START cloud_sql_postgres_instance_psc]
+resource "google_sql_database_instance" "postgres_instance_psc" {
```
Comment (rogerthatdev) [inline cloud_sql/instance_psc/main.tf:41]: After splitting the file, rename each `google_sql_database_instance` to `default` per samples' best practice guidelines.

**Review by rogerthatdev — CHANGES_REQUESTED** [review]
Let's split this sample like the other cloud_sql samples

**Comment by msampathkumar** [comment]
As mentioned by @rogerthatdev, kindly use DB specific folder prefixes.

* mysql_instance_...
* postgres_instance_...

https://github.com/terraform-google-modules/terraform-docs-samples/tree/main/cloud_sql

File: cloud_sql/instance_psc/main.tf
Comment (isaurabhuttam) [reply cloud_sql/instance_psc/main.tf:1]: Done.

**Comment by msampathkumar** [comment]
/gcbrun

Re-triggering the CICD pipeline

File: cloud_sql/mysql_instance_psc/main.tf [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# [START cloud_sql_mysql_instance_psc]
+resource "google_sql_database_instance" "main" {
```
Comment (msampathkumar) [inline cloud_sql/mysql_instance_psc/main.tf:18]:
```suggestion
resource "google_sql_database_instance" "default" {
```

Kindly use `default` as the default name for resource.

File: cloud_sql/mysql_instance_psc/main.tf [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      }
+      ipv4_enabled = false
+    }
+  }
+}
+# [END cloud_sql_mysql_instance_psc]
+
+# [START cloud_sql_mysql_instance_psc_endpoint]
+resource "google_compute_address" "main" {
+  name         = "psc-compute-address-${google_sql_database_instance.main.name}"
```
Comment (msampathkumar) [inline cloud_sql/mysql_instance_psc/main.tf:42]:
```suggestion
resource "google_compute_address" "default" {
  name         = "psc-compute-address-${google_sql_database_instance.default.name}"
```

File: cloud_sql/mysql_instance_psc/main.tf [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  address_type = "INTERNAL"
+  subnetwork   = "default"     # Replace value with the name of the subnet here.
+  address      = "10.128.0.43" # Replace value with the IP address to reserve.
+}
+
+data "google_sql_database_instance" "main" {
+  name = resource.google_sql_database_instance.main.name
```
Comment (msampathkumar) [inline cloud_sql/mysql_instance_psc/main.tf:49]:
```suggestion
data "google_sql_database_instance" "default" {
  name = resource.google_sql_database_instance.default.name
```

**Review by rogerthatdev — APPROVED** [review]
After a few minor changes, these samples individually pass int tests.

**Pull request merged**